src/huggingface_hub/cli/cache.py

```diff
@@ -12,368 +12,629 @@
         ),
     ] = None,
-    verbose: Annotated[
-        int,
+    revisions: Annotated[
+        bool,
+        typer.Option(
+            help="Include revisions in the output instead of aggregated repositories.",
+        ),
+    ] = False,
+    filter: Annotated[
+        Optional[list[str]],
+        typer.Option(
+            "-f",
+            "--filter",
+            help="Filter entries (e.g. 'size>1GB', 'type=model', 'accessed>7d'). Can be used multiple times.",
+        ),
+    ] = None,
+    format: Annotated[
+        OutputFormat,
+        typer.Option(
+            help="Output format.",
+        ),
+    ] = OutputFormat.table,
+    quiet: Annotated[
+        bool,
         typer.Option(
-            "-v",
-            "--verbose",
-            count=True,
-            help="Increase verbosity (-v, -vv, -vvv).",
+            "-q",
+            "--quiet",
+            help="Print only IDs (repo IDs or revision hashes).",
         ),
-    ] = 0,
+    ] = False,
 ) -> None:
     try:
-        t0 = time.time()
-        hf_cache_info = scan_cache_dir(dir)
-        t1 = time.time()
+        hf_cache_info = scan_cache_dir(cache_dir)
     except CacheNotFound as exc:
         print(f"Cache directory not found: {str(exc.cache_dir)}")
+        raise typer.Exit(code=1)
+
+    filters = filter or []
+
+    entries, repo_refs_map = collect_cache_entries(hf_cache_info, include_revisions=revisions)
+    try:
+        filter_fns = [compile_cache_filter(expr, repo_refs_map) for expr in filters]
+    except ValueError as exc:
+        raise typer.BadParameter(str(exc)) from exc
+
+    now = time.time()
+    for fn in filter_fns:
+        entries = [entry for entry in entries if fn(entry[0], entry[1], now)]
+
+    if quiet:
+        for repo, revision in entries:
+            print(revision.commit_hash if revision is not None else format_cache_repo_id(repo))
         return
-    print(get_table(hf_cache_info, verbosity=verbose))
-    print(
-        f"\nDone in {round(t1 - t0, 1)}s. Scanned {len(hf_cache_info.repos)} repo(s)"
-        f" for a total of {ANSI.red(hf_cache_info.size_on_disk_str)}."
-    )
-    if len(hf_cache_info.warnings) > 0:
-        message = f"Got {len(hf_cache_info.warnings)} warning(s) while scanning."
-        if verbose >= 3:
-            print(ANSI.gray(message))
-            for warning in hf_cache_info.warnings:
-                print(ANSI.gray(str(warning)))
-        else:
-            print(ANSI.gray(message + " Use -vvv to print details."))
 
+    formatters = {
+        OutputFormat.table: print_cache_entries_table,
+        OutputFormat.json: print_cache_entries_json,
+        OutputFormat.csv: print_cache_entries_csv,
+    }
+    return formatters[format](entries, include_revisions=revisions, repo_refs_map=repo_refs_map)
 
-@cache_cli.command("delete", help="Delete revisions from the cache directory")
-def cache_delete(
-    dir: Annotated[
+
+@cache_cli.command(help="Remove cached repositories or revisions.")
```

Comment:
```suggestion
@cache_cli.command()
```

feel more "typer-ic" to use docstring here